Jakefile.js

```diff
@@ -921,92 +505,123 @@ function runConsoleTests(defaultReporter, runInParallel) {
         });
         lint.invoke();
     }
+
+    function deleteTemporaryProjectOutput() {
+        if (fs.existsSync(path.join(Paths.baselines.local, "projectOutput/"))) {
+            jake.rmRf(path.join(Paths.baselines.local, "projectOutput/"));
+        }
+    }
 }
 
-desc("Runs all the tests in parallel using the built run.js file. Optional arguments are: t[ests]=category1|category2|... d[ebug]=true.");
-task("runtests-parallel", ["build-rules", "tests", builtLocalDirectory, run], function () {
-    runConsoleTests('min', /*runInParallel*/ true);
-}, { async: true });
+// used to pass data from jake command line directly to run.js
+function writeTestConfigFile(tests, runners, light, taskConfigsFolder, workerCount, stackTraceLimit, colors, testTimeout) {
+    var testConfigContents = JSON.stringify({
+        runners: runners ? runners.split(",") : undefined,
+        test: tests ? [tests] : undefined,
+        light: light,
+        workerCount: workerCount,
+        taskConfigsFolder: taskConfigsFolder,
+        stackTraceLimit: stackTraceLimit,
+        noColor: !colors,
+        timeout: testTimeout
+    });
+    fs.writeFileSync('test.config', testConfigContents, { encoding: "utf-8" });
+}
 
-desc("Runs the tests using the built run.js file. Optional arguments are: t[ests]=regex r[eporter]=[list|spec|json|<more>] d[ebug]=true color[s]=false lint=true bail=false dirty=false.");
-task("runtests", ["build-rules", "tests", builtLocalDirectory, run], function () {
-    runConsoleTests('mocha-fivemat-progress-reporter', /*runInParallel*/ false);
-}, { async: true });
+function cleanTestDirs() {
+    // Clean the local baselines directory
+    if (fs.existsSync(Paths.baselines.local)) {
+        del.sync(Paths.baselines.local);
+    }
 
-desc("Generates code coverage data via instanbul");
-task("generate-code-coverage", ["tests", builtLocalDirectory], function () {
-    var testTimeout = process.env.timeout || defaultTestTimeout;
-    var cmd = 'istanbul cover node_modules/mocha/bin/_mocha -- -R min -t ' + testTimeout + ' ' + run;
-    console.log(cmd);
-    exec(cmd);
-}, { async: true });
+    // Clean the local Rwc baselines directory
+    if (fs.existsSync(Paths.baselines.localRwc)) {
+        del.sync(Paths.baselines.localRwc);
+    }
 
-// Browser tests
-var nodeServerOutFile = "tests/webTestServer.js";
-var nodeServerInFile = "tests/webTestServer.ts";
-compileFile(nodeServerOutFile, [nodeServerInFile], [builtLocalDirectory, tscFile], [], /*useBuiltCompiler:*/ true, { noOutFile: true, lib: "es6" });
+    jake.mkdirP(Paths.baselines.local);
+    jake.mkdirP(Paths.baselines.localRwc);

```

Comment:
This is going to make an `internal` folder for people without `internal` checked out. We probably don't want that.